### DISCUSSION TITLE
User update any value in the batch details, the batch start date is auto updated and the date is reduced by 1 from the existing date.

### DISCUSSION THREAD

**Discussion by kamal-singhal** [Sunbird-Lern/Community#176] [Issues]
Hi team,

We are facing issue that Whenever the user update any value in the course batch details, the batch start date is auto updated and the value is reduced by 1 from the existing date in the DB.
Database details:
server: cassandra
keyspace: sunbird_course
table name : course_batch

We also checked the batch start date in the another table as content_hierarchy, in this table date of start date showing fine as expected.
Can you help us to figure out what is the issue that why start date is auto update whenever user update batch details. On the report the batch start date coming from the course batch table so it's showing wrong date on the report.

We need your help to debug this issue. for your reference, attached the screen shots of UI & data in the database for for your better understanding.
![DB_data_course_batch](https://github.com/Sunbird-Lern/Community/assets/125978084/4df0fd54-df57-479e-99fd-f098a0d917dd)
![Batch_Update_console](https://github.com/Sunbird-Lern/Community/assets/125978084/7d6508aa-f5a4-4c53-8e84-15c8b2dde091)

**Comment by Jayaprakash8887** [comment]
@kamal-singhal Please share the CURL that is getting triggered on 'Update' button.

**Comment by yogeshsarmalkar-trigyn** [comment]
@Jayaprakash8887 Please find the below curl request which is called during update

curl --location --request PATCH 'https://dev.oci.diksha.gov.in/learner/course/v1/batch/update' \
--header 'Accept: application/json' \
--header 'Accept-Language: en-US,en;q=0.9,hi;q=0.8' \
--header 'Connection: keep-alive' \
--header 'Content-Type: application/json' \
--header 'Cookie: _ga=GA1.1.1057168182.1701153996; connect.sid=s%3A-iZ0DLV_QWzokCfyrD1bQD6pb_CRSUk0.TKdQ76y5pX0F2LPQExleFpP8fK0zmtb5yC3BTf%2Bokf0; _ga_9Z3LDJ42BE=GS1.1.1701859363.28.1.1701859912.0.0.0' \
--header 'Origin: https://dev.oci.diksha.gov.in' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-origin' \
--header 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36' \
--header 'X-App-Id: dev.diksha.portal' \
--header 'X-App-Version: 5.1.0' \
--header 'X-Channel-Id: 0125196274181898243' \
--header 'X-Device-ID: 35b3bf5f0ce7fb8b62c664cc9e123150' \
--header 'X-Org-code: 0125196274181898243' \
--header 'X-Request-ID: 4d4c3889-1a40-8e53-6d40-82dee783c730' \
--header 'X-Session-ID: -iZ0DLV_QWzokCfyrD1bQD6pb_CRSUk0' \
--header 'X-Source: web' \
--header 'X-User-ID: e2419c04-2862-44da-a1c2-28d9ab5d641f' \
--header 'X-msgid: 4d4c3889-1a40-8e53-6d40-82dee783c730' \
--header 'sec-ch-ua: "Google Chrome";v="119", "Chromium";v="119", "Not?A_Brand";v="24"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "Windows"' \
--header 'ts: 2023-12-06T16:23:35+05:30' \
--data-raw '{
    "request": {
        "id": "01394163838550016010",
        "courseId": "do_3139416330257530881299",
        "name": "new batch ",
        "description": "new batch 1",
        "enrollmentType": "open",
        "startDate": "2023-12-06",
        "endDate": "2023-12-21",
        "createdFor": [
            "0125196274181898243"
        ],
        "mentors": [],
        "enrollmentEndDate": "2023-12-13"
    }
}'

**Reply by Jayaprakash8887** [reply]
@yogeshsarmalkar-trigyn Is the payload carrying the faulty data to the Batch Update API ? If yes, issue needs to be verified and addressed in front-end service by Sunbird-ED BB team. If the payload is proper and the backend service is modifying the data (start-date) unexpectedly, then please share the course-batch service logs.
